ops/ruby/devops-framework/DEVOPS_FRAMEWORK.md

```diff
@@ -0,0 +1,78 @@
+# DevOps Framework
+
+## Purpose
+This is a small but flexible framework for automating DevOps actions and ensuring they are
+taken in a traceable, repeatable, and consistent way. It operates by  communicating to Google Cloud Platform
+directly, independently of the existing build, operations, and deployment pipeline(s).
+
+## Usage
+The entry point to all functionality is `devops.rb`, which provides command line and environment
+loading and dispatches to Task classes (not scripts) that do the real work.
+`./devops.rb  -t <task_name> -e <env_file> --task-option-1 value1`
+
+There is also a dynamically-generated help file from the descriptions given in the options.
+```
+$ ruby ./devops.rb  --help
+Usage: devops [options]
+    -t, --task [TASK]                Task to be in in each environment
+    -e, --envs-file [ENVS_FILE]      Path to environments JSON file.
+```
+
+##  Installation
+We need a newish version of Ruby. I'm running 2.6.5 as of this writing, but haven't gone
+back and tested with older versions. We also currently need the `gcloud` tools and an account with
+sufficient privilege account to create service account tokens.
+
+Additionally, you'll need to install a gem or two. Right now, it's
+
+`gem install google-cloud-monitoring` for the monitoring library
```

Comment:
First attempt: you don't have permission to write to your gems folder

Second attempt: install ruby via Homebrew and update paths
```
ERROR:  Error installing google-cloud-monitoring:
        The last version of grpc (~> 1.24) to support your Ruby & RubyGems was 1.27.0. Try installing it with `gem install grpc -v 1.27.0` and then running the current command again
        grpc requires Ruby version >= 2.3, < 2.7.dev. The current ruby version is 2.7.0.0.

$ ruby --version
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-darwin18]

$ gem install grpc -v 1.27.0
ERROR:  Error installing grpc:
        The last version of grpc (= 1.27.0) to support your Ruby & RubyGems was 1.27.0. Try installing it with `gem install grpc -v 1.27.0`
        grpc requires Ruby version >= 2.3, < 2.7.dev. The current ruby version is 2.7.0.0.
```

Third attempt: downgrade Homebrew ruby to 2.6 ... nope.  not available!  2.5 then.
```
$ brew install ruby@2.5
```
update my PATH again
```
$ ruby --version
ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-darwin18]
$ gem install google-cloud-monitoring
```

Success, finally.

Reply:
The command given here also failed. This worked for me:

```
gem install --user google-cloud-monitoring
```

We should probably use a Gemfile if we're going to start requiring gem installation, rather than trying to keep this README up to date.

Reply:
Rather than these approaches, my strong preference would be to dockerize this so modifying your local Ruby install is not necessary.

Reply:
That's interesting. we can do a gemfile first I think. I don't know if I have a dockerization budget; I'm barely going to get away with doing this much.

We probably need to document local ruby anyway for build stuff.